include/filepath.h

```diff
@@ -94,9 +94,22 @@ class Filepath {
 	// Return `false` and do nothing if Filepath is empty, set extension and
 	// return `true` otherwise.
 	//
+	// See also `add_extension()`.
+	//
 	// \note `ext` is interpreted as bytes in locale encoding.
 	bool set_extension(const std::string& ext);
 
+	// Return `false` and do nothing if Filepath is empty, append extension and
+	// return `true` otherwise.
+	//
+	// The difference from `set_extension()` is that `set_extension()` replaces
+	// the existing extension while `add_extension()` appends to it. For the
+	// path `foo.tar`, `set_extension("gz")` will turn the path into "tar.gz",
```

Comment:
> For the path `foo.tar`, `set_extension("gz")` will turn the path into "tar.gz"

That sounds very surprising.
Shouldn't that be `turn the path into "foo.gz"`?

Reply:
You're right, that's a mistake.